From ed16e20bc07ecf365f399f9c801cd01ed5cda36c Mon Sep 17 00:00:00 2001
From: startergo <7897244+startergo@users.noreply.github.com>
Date: Wed, 24 Jun 2026 11:16:57 -0500
Subject: [PATCH 18/36] [sl-port-610] Fix control-paint crash:
 ThemeMac::ensuredView called +[NSAppearance currentAppearance] /
 -setAppearance: (10.9+) unguarded -> SIGTRAP in _class_initialize on 10.6
 when painting any themed control (toggle/button). Guard with
 NSClassFromString + respondsToSelector:; 10.6 has only default Aqua

---
 Source/WebCore/platform/mac/ThemeMac.mm | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/mac/ThemeMac.mm b/Source/WebCore/platform/mac/ThemeMac.mm
index 941f72f31c..77671202ac 100644
--- a/Source/WebCore/platform/mac/ThemeMac.mm
+++ b/Source/WebCore/platform/mac/ThemeMac.mm
@@ -720,7 +720,12 @@ NSView *ThemeMac::ensuredView(ScrollView* scrollView, const ControlStates& contr
     // Use a fake view.
     static WebCoreThemeView *themeView = [[WebCoreThemeView alloc] init];
     [themeView setFrameSize:NSSizeFromCGSize(scrollView->totalContentsSize())];
-    [themeView setAppearance:[NSAppearance currentAppearance]];
+    // [leopard] NSAppearance / -setAppearance: / +currentAppearance are 10.9+.
+    // On 10.6 NSAppearance does not exist; touching +[NSAppearance currentAppearance]
+    // traps during class init (SIGTRAP in _class_initialize). 10.6 has only the
+    // default Aqua appearance, so skip setting it.
+    if ([themeView respondsToSelector:@selector(setAppearance:)] && NSClassFromString(@"NSAppearance"))
+        [themeView setAppearance:[NSAppearance currentAppearance]];
 
     themeWindowHasKeyAppearance = !(controlStates.states() & ControlStates::WindowInactiveState);
 
-- 
2.54.0

